src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int forNullKeyIndex = 0;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private int capacity;
+    private MapNode<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        table = createTable(capacity);
+        threshold = (int) (capacity * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        MapNode<K, V> toPut = new MapNode<>(key, value);
+        MapNode<K, V> currentNode = table[calculateIndex(toPut)];
+        int index = calculateIndex(toPut);
+        if (currentNode == null) {
+            table[index] = toPut;
+            size++;
+        } else {
+            while (true) {
+                if (currentNode.key == null && key == null) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.equals(toPut)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = toPut;
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = calculateIndex(new MapNode<>(key, null));
+        MapNode<K, V> currentNode = table[index];
+
+        while (currentNode != null) {
+
+            if (currentNode.key != null && currentNode.key.equals(key)
+                    || currentNode.key == null && key == null) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int calculateIndex(MapNode<K, V> node) {
+        if (node == null) {
+            return forNullKeyIndex;
```

Comment:
you shouldn't handle null keys separately. Always use null safe comparison like Objects.equals() and you won't need it